### PR TITLE
[BEAM-1704] Added Create.TimestampedValues.withType

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @tgroh

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java [hunk 1]
```diff
@@ -303,6 +303,30 @@ public void testCreateTimestampedPolymorphicType() throws Exception {
   }
 
   @Test
+  public void testCreateTimestampedDefaultOutputCoderUsingCoder() throws Exception {
+    Coder<Record> coder = new RecordCoder();
+    PBegin pBegin = PBegin.in(p);
+    Create.TimestampedValues<Record> values =
+        Create.timestamped(TimestampedValue.of(new Record(), new Instant()),
+            TimestampedValue.of(new Record(), new Instant())).withCoder(coder);
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java]: one of these should be `new Record2()`

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java [hunk 2]
```diff
@@ -303,6 +303,30 @@ public void testCreateTimestampedPolymorphicType() throws Exception {
...
+  }
+
+  @Test
+  public void testCreateTimestampedDefaultOutputCoderUsingTypeDescriptor() throws Exception {
+    Coder<Record> coder = new RecordCoder();
+    p.getCoderRegistry().registerCoder(Record.class, coder);
+    PBegin pBegin = PBegin.in(p);
+    Create.TimestampedValues<Record> values =
+        Create.timestamped(TimestampedValue.of(new Record(), new Instant()),
+            TimestampedValue.of(new Record(), new Instant()))
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/CreateTest.java]: Likewise.

**Comment by aviemzur** [comment]
@tgroh made the required changes in tests. PTAL

**Comment by aviemzur** [comment]
@tgroh made the required changes in tests. PTAL

**Pull request closed without merging**